src/green/gpu/cugw_qpt.h

```diff
@@ -374,8 +402,20 @@ namespace green::gpu {
 
     // pointer to cublas handle
     cublasHandle_t* handle_;
+
+    // status of data transfer / copy from Device to Host.
+    // false: not required, stream ready for next calculation
+    // true: required
+    bool cleanup_req_;
```

Comment:
propose making a destructor that throws an exception if cleanup_req_ is true and a constructor that sets it to false.

Reply:
That's right. I guess we need to print an error and abort the program. This should never happen unless there's a logic error anyway.